base/logging.go

```diff
@@ -10,16 +10,95 @@
 package base
 
 import (
+	"errors"
 	"fmt"
+	"github.com/couchbase/clog"
+	"github.com/natefinch/lumberjack"
 	"log"
 	"os"
 	"runtime"
 	"strings"
 	"sync"
 	"time"
-	"github.com/couchbase/clog"
 )
 
+var errMarshalNilLevel = errors.New("can't marshal a nil *Level to text")
+
+type Level int32
+
+const (
+	// DebugLevel logs are typically voluminous, and are usually disabled in
+	// production.
+	DebugLevel Level = iota - 1
```

Comment:
This was taken from [the zap logging package](https://github.com/uber-go/zap/blob/master/level.go). Note: The code at zap has changed slightly but the level values have not.

By setting debug to -1, if log level is not set in the logging config it will default to the zero value for int32 (0) which will disable debug logging, info logging will be the default output.

I have also added this comment to the source.